File: 1537-maximum-score-after-splitting-a-string/maximum-score-after-splitting-a-string.py
class Solution:
    def maxScore(self, s: str) -> int:

        dicti={}
        count=0
        for i in s:
            if i=='1':
                dicti[i]=dicti.get(i,0)+1
        res=0
        # if len(dicti)==0:
        #     return 1
        for i in range(len(s)-1):
            if s[i]=='0':
                count+=1
            else:
                if s[i] in dicti and dicti[s[i]]>0:
                    dicti[s[i]]-=1
            res=max(res,(count + dicti.get('1',0)))
        return res
            
                
        

File: 1537-maximum-score-after-splitting-a-string/maximum-score-after-splitting-a-string.js
/**
 * @param {string} s
 * @return {number}
 */
var maxScore = function(s) {

    let dict= {}    
    max=0;
    for(let i=0;i<s.length;i++)
    {
        if(s[i]==='1')
        {
            dict[s[i]]=(dict[s[i]] || 0)+1;
        }
    }
    let count=0;

    for(let i=0;i<s.length-1;i++)
    {
        if(s[i]==='0')
        {
            count++;
        }
        else{
            if(dict[s[i]] && dict[s[i]]>0 )
            {
                dict[s[i]]-=1;
            }
        }

        max=Math.max(max,count+(dict['1'] || 0));
        
    }

    return max;
    
};